### PR TITLE
Add Question Service

### LINKED ISSUE #14: F2.1.1 The service should expose an endpoint for the admin to create new questions and include the 5 compulsory attributes (Id, Title, Description, Category, Complexity).

### LINKED ISSUE #50: F2.2.1 The service should expose an endpoint to allow users to retrieve the list of available questions.

### PR DESCRIPTION
**Pull request by dylkaw**
This PR resolves #14 and resolves #50.

## Overview
This PR sets up the base for the FastAPI Question Service. In this PR, basic Create and Read endpoints have been created.

## Setup
1. In the `question-service` directory, run `python -m venv .venv` to create a virtual environment.
2. Run `source .venv/bin/activate` to activate the virtual environment.
3. Run `pip install -r requirements.txt` to install all necessary dependencies. Use Python 3.12.
4. Ensure you have a `.env` file with the question service mongoDB uri. 
5. To start the service, run `fastapi dev app/main.py`

Note: For VSCode to properly detect the venv, open the project with `question-service` as the root directory. If not, VSCode will not detect the venv properly and will scream at you for everything. 

## Backend changes
1. `'QuestionModel` created, it has fields title, description, category and complexity. id is not necessary as it is automatically generated by mongoDB on create.
2. `create_question` and `get_questions` endpoints created. 

Question service will be served at `http://127.0.0.1:8000` and you will be able to test requests directly on the Swagger UI at `http://127.0.0.1:8000/docs`.

### CODE REVIEW

**Comment by wr1159** [comment]
Can you add a README.md for how to set it up? would be greatly appreciated thanks

File: question-service/requirements.txt [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+python-multipart==0.0.9
+PyYAML==6.0.2
+rich==13.8.1
+shellingham==1.5.4
+sniffio==1.3.0
+starlette==0.38.5
+typer==0.12.5
+typing_extensions==4.8.0
+uvicorn==0.28.0
+uvloop==0.20.0
```
Comment (jq1836) [inline question-service/requirements.txt:33]: Need to remove uvloop, otherwise wouldn't work on Windows OS

**Review by jq1836 — CHANGES_REQUESTED** [review]
Require some minor changes to support Windows OS. Apart from that, LGTM!

File: question-service/README.md [hunk 2]
```diff
@@ -0,0 +1,10 @@
+Dependencies: Python 12
+
+## Setup
+1. In the `question-service` directory, run `python -m venv .venv` to create a virtual environment.
+2. Run `source .venv/bin/activate` to activate the virtual environment.
```
Comment (jq1836) [inline question-service/README.md:5]: Btw, this is different for Windows, supposed to just run .venv/Scripts/activate, apart from that, its the same

**Review by SelwynAng — COMMENTED** [review]
Perhaps can follow a project structure as shown here https://medium.com/@amirm.lavasani/how-to-structure-your-fastapi-projects-0219a6600a8f

**Review by SelwynAng — APPROVED** [review]
LGTM!

**Review by jq1836 — APPROVED** [review]
LGTM!

**Pull request merged**